src/main/ruby/truffleruby/core/process.rb

```diff
@@ -435,7 +435,14 @@ def self.euid=(uid)
   end
 
   def self.egid=(gid)
-    gid = Truffle::Type.coerce_to gid, Integer, :to_int
+    gid =
+      if gid.kind_of?(String)
+        require 'etc'
+        Etc.getgrnam(gid).gid
+      else
+        Truffle::Type.coerce_to gid, Integer, :to_int
+      end
```

Comment:
To match the type conversion here: https://github.com/ruby/ruby/blob/0db68f023372b634603c74fca94588b457be084c/process.c#L6006

Maybe this could be something like:
```ruby
    gid =
      if Truffle::Type.fits_into_long?(gid) ||
        (tmp = Truffle::Type.rb_check_convert_type(gid, String, :to_str)).nil?
        Truffle::Type.rb_num2uint(gid)
      else
        require 'etc'
        Etc.getgrnam(tmp).gid
      end
```